xbmc/input/linux/LIRC.cpp

```diff
@@ -181,11 +201,17 @@ void CRemoteControl::Update()
 
   uint32_t now = XbmcThreads::SystemClockMillis();
 
+  char buf[128];
   // Read a line from the socket
-  while (fgets(m_buf, sizeof(m_buf), m_file) != NULL)
+  while (true)
   {
+    CSingleLock lock(m_CS);
+    if (fgets(buf, sizeof(buf), m_file) == NULL)
+      break;
+    lock.unlock();
```

Comment:
I always try to avoid those by using an extra block for the section which requires the lock.